app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -32,6 +41,49 @@ public LoginPresenter() {
 
     }
 
+    /**
+     *
+     * @param email EMail
+     * @param password Pass
+     */
+    public void onLoginButtonClicked(@Nullable String email, @Nullable String password) {
+
+        if (validateFields(email, password)) {
+
+            UserService usuarioService = RetrofitInstance.getRetrofitInstance().create(UserService.class);
+            Call<List<User>> call = usuarioService.getUserLogin(email);
+
+            call.enqueue(new NetworkCallback<List<User>>() {
+
+                @Override
+                public void onResponseSuccessful(@Nullable List<User> response) {
+
+                    if ((response.isEmpty()) || (!response.get(0).getPassword().equals(password))) {
+                        getView().showLoginFailure(R.string.login_error_credentials);
+                    } else {
+                        editor.putString(mPrefEmail, email);
+                        editor.putString(mPrefPass, password);
+                        editor.commit();
+
+                        getView().onLoginButtonPressed();
```

Comment:
It's not ok to call this method onLoginButtonPressed. Fragment should be a silly class, however Presenter won't. This method should be called onLoginSuccesfully. "Copy" what you did in onResponseFailed/onCallFailure.